packages/hmr-plugin/src/internal/state-event-loop.ts

```diff
@@ -0,0 +1,43 @@
+import { NgxsBootstrapper, StateContext, StateStream } from '@ngxs/store';
+import { NgModuleRef } from '@angular/core';
+import { Subscription } from 'rxjs';
+
+import { getStateContext } from './state-context';
+import { CallStackFrame, NgxsHmrLifeCycle } from '../symbols';
+import { getBootstrap, getStateFromHmrStorage, getStateStream } from './common';
+
+export function stateEventLoop<T extends NgxsHmrLifeCycle<S>, S>(
+  ref: NgModuleRef<T>,
+  frame: CallStackFrame<S>
+) {
+  const stateContext: StateContext<S> | undefined = getStateContext<T, S>(ref);
+  if (stateContext) {
+    const previousState: Partial<S> = getStateFromHmrStorage<S>();
+    const existSavedState: boolean = Object.keys(previousState).length > 0;
```

Comment:
```suggestion
    const savedStateExists: boolean = Object.keys(previousState).length > 0;
```
Proposed variable rename.

Reply:
@markwhitfeld maybe just `const savedStateExists: boolean = Object.keys(previousState).length`, I mean without `> 0`?

Reply:
Then its type would not be boolean. I think it is best to leave the `> 0` part there.

Reply:
> Then its type would not be boolean. I think it is best to leave the `> 0` part there.

So he can add `!!` in the beginning.

Reply:
You could do the `!!` but it offers no advantage over the `> 0` which actually describes what we are doing and requires no translation.